pkg/apis/extensions/types.go

```diff
@@ -818,3 +819,82 @@ type PodSecurityPolicyList struct {
 
 	Items []PodSecurityPolicy `json:"items"`
 }
+
+// Template contains the inputs needed to produce a Config.
+type Template struct {
+	unversioned.TypeMeta `json:",inline"`
+	api.ObjectMeta       `json:"metadata,omitempty"`
+
+	Spec TemplateSpec `json:"spec,omitempty"`
+}
+
+type TemplateSpec struct {
+	// Optional: Parameters is an array of Parameters used during the
+	// Template to Config transformation.
+	Parameters []Parameter `json:"parameters,omitempty"`
+
+	// Required: A yaml config list of resources to create
+	Items []runtime.RawExtension `json:"items,omitempty"`
```

Comment:
runtime.Object won't work because trying to parse the items into actual go objects will fail.  (Trying to parse TEMPLATE_PARAM_NAME into an int field would fail)